src/UKHO.ADDS.EFS.Builder.S100/Pipelines/Distribute/UploadFilesNode.cs

```diff
@@ -1,10 +1,92 @@
+        private ILogger _logger;
+
+        private const int FileBufferSize = 81920;
+
+        public UploadFilesNode(IFileShareReadWriteClient fileShareReadWriteClient) : base()
+        {
+            _fileShareReadWriteClient = fileShareReadWriteClient ?? throw new ArgumentNullException(nameof(fileShareReadWriteClient));
+        }
+
+        protected override async Task<NodeResultStatus> PerformExecuteAsync(IExecutionContext<ExchangeSetPipelineContext> context)
+        {
+            _logger = context.Subject.LoggerFactory.CreateLogger<UploadFilesNode>();
+            var batchId = context.Subject.BatchId;
+            var correlationId = context.Subject.Job.CorrelationId;
+            var jobId = context.Subject.Job?.Id;
+
+            string fileName = context.Subject.ExchangeSetFileName;
+            string filePath = Path.Combine(context.Subject.ExchangeSetFilePath, context.Subject.ExchangeSetOutputDirectory, $"{jobId}.zip");
+
+            if (!File.Exists(filePath))
+            {
+                _logger.LogAddFileNodeFailed($"File not found at given path for Job Id: {jobId}");
+                return NodeResultStatus.Failed;
+            }
+
+            try
+            {
+                await using var fileStream = GetExchangeSetFileStream(filePath);
+
+                var batchHandle = new BatchHandle(batchId);
+                var addFileResult = await _fileShareReadWriteClient.AddFileToBatchAsync(
+                    batchHandle,
+                    fileStream,
+                    fileName,
+                    ApiHeaderKeys.ContentTypeOctetStream,
+                    correlationId,
+                    CancellationToken.None
+                );
+
+                if (!addFileResult.IsSuccess(out _, out var error))
+                {
+                    LogAddFileFailure(fileName, batchId, correlationId, error);
+                    return NodeResultStatus.Failed;
+                }
+
+                return NodeResultStatus.Succeeded;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogAddFileNodeFailed(ex.Message);
+                return NodeResultStatus.Failed;
+            }
+        }
+
+        private static FileStream GetExchangeSetFileStream(string filePath)
+        {
+            return new FileStream(
+                filePath,
+                FileMode.Open,
+                FileAccess.Read,
+                FileShare.Read,
+                FileBufferSize,
+                FileOptions.Asynchronous | FileOptions.SequentialScan);
+        }
+
+        private void LogAddFileFailure(string fileName, string batchId, string correlationId, IError error)
+        {
+            var addFileLogView = new AddFileLogView
+            {
+                FileName = fileName,
+                BatchId = batchId,
+                CorrelationId = correlationId,
+                Error = error
+            };
+
+            _logger.LogAddFileNodeFssAddFileFailed(addFileLogView);
```

Comment:
Use a clearer, more concise method name
```suggestion
            _logger.LogFileShareAddFileToBatchError(addFileLogView);
```

Reply:
Renamed as suggested.